crates/biome_js_analyze/src/aria_analyzers/nursery/use_valid_aria_role.rs

```diff
@@ -0,0 +1,125 @@
+use crate::aria_services::Aria;
+use biome_analyze::{context::RuleContext, declare_rule, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::jsx_ext::AnyJsxElement;
+use biome_rowan::AstNode;
+use serde::{Deserialize, Serialize};
+
+declare_rule! {
+    /// Elements with ARIA roles must use a valid, non-abstract ARIA role.
+    ///
+    /// Source: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/aria-role.md
+    ///
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role="datepicker"></div>
+    ///  ```
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role="range"></div>
+    ///  ```
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role=""></div>
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    ///   <Foo role="foo"></Foo>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// <>
+    ///   <div role="button"></div>
+    ///   <div role={role}></div>
+    ///   <div></div>
+    /// </>
+    /// ```
+    ///
+    /// ## Accessibility guidelines
+    /// - [WCAG 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)
+    ///
+    /// ## Resources
+    /// - [Chrome Audit Rules, AX_ARIA_01](https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_01)
+    /// - [DPUB-ARIA roles](https://www.w3.org/TR/dpub-aria-1.0/)
+    /// - [MDN: Using ARIA: Roles, states, and properties](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques)
+    ///
+    pub(crate) UseValidAriaRole {
+        version: "next",
+        name: "useValidAriaRole",
+        recommended: false,
+    }
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct UseValidAriaRoleOptions {
+    allowed_invalid_roles: Vec<String>,
+    ignore_non_dom: bool,
+}
+
+impl Rule for UseValidAriaRole {
+    type Query = Aria<AnyJsxElement>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = UseValidAriaRoleOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let options = ctx.options();
+        let aria_roles = ctx.aria_roles();
+
+        let ignore_non_dom = options.ignore_non_dom;
+        let allowed_invalid_roles = &options.allowed_invalid_roles;
+
+        if ignore_non_dom && node.is_custom_component() {
+            return None;
+        }
+
+        let role_attribute = node.find_attribute_by_name("role")?;
+        if role_attribute.is_value_null_or_undefined() {
+            return None;
+        }
+
+        if node.is_custom_component() {
+            return Some(());
+        }
```

Comment:
I think these lines are not necessary. We need to check the role for the custom component as well.